### PR TITLE
Rpc <> Exex example

### LINKED ISSUE #31: Add an example of using an RPC with an ExEx
### Describe the feature

Add an example of how to start a custom RPC method alongside an ExEx, and feed the data from the RPC method into the ExEx, and from the ExEx into the RPC response.

The ExEx does not need to do anything super special; it could e.g. something like the optimism deposit decoder example we have that just also exposes # deposits over RPC or similar. The goal is to demonstrate:

1. How to add a custom RPC method for an ExEx
2. Feed data from that RPC method to an ExEx
3. Feed data from an ExEx to an RPC response

### Additional context

_No response_

### LINKED ISSUE #8008: Add an example of using an RPC with an ExEx
### Describe the feature

Add an example of how to start a custom RPC method alongside an ExEx, and feed the data from the RPC method into the ExEx, and from the ExEx into the RPC response.

The ExEx does not need to do anything super special; it could e.g. something like the optimism deposit decoder example we have that just also exposes # deposits over RPC or similar. The goal is to demonstrate:

1. How to add a custom RPC method for an ExEx
2. Feed data from that RPC method to an ExEx
3. Feed data from an ExEx to an RPC response

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by supernovahs**
closes paradigmxyz/reth-exex-examples#31

I am trying to write a custom rpc to calculate the no of deposits in op bridge . But getting some errors. Can u point me to correct it. Thanks

### CODE REVIEW

**Comment by supernovahs** [comment]
@onbjerg  @mattsse

File: examples/exex/custom-rpc/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
+`
```
Comment (onbjerg) [inline examples/exex/custom-rpc/src/main.rs:1]:
```suggestion

```

File: examples/exex/custom-rpc/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                Err(_) =>{
+                   Ok(0)
+                }
+            }
+        },
+            Err(_)=>{
+               Ok(0)
+        }
+
+       }
```
Comment (onbjerg) [inline examples/exex/custom-rpc/src/main.rs:109]: this is not correct, we want this to ask the exex what the deposit count is, we don't want to calculate it ourselves on each call

File: examples/exex/custom-rpc/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+                    println!("txpool extension enabled");
+
+                    Ok(())
+                })
+                .launch()   
+                .await?;
+
+            handle.wait_for_node_exit().await
+        })
```
Comment (onbjerg) [inline examples/exex/custom-rpc/src/main.rs:41]: the exex is missing, pls see the other examples

File: examples/exex/custom-rpc/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    #[arg(long)]
+    pub enable_ext: bool,
+}
+
+/// trait interface for a custom rpc namespace: `opdepositcount`
+///
+/// This defines an additional namespace where all methods are configured as trait functions.
+#[rpc[server, namespace="onDepositCount"]]
+pub trait OpDepositCountExtApi {
+    #[method(name = "opdepositCount")]
```
Comment (onbjerg) [inline examples/exex/custom-rpc/src/main.rs:57]: please follow the naming convention of the other namespaces

**Review by onbjerg — CHANGES_REQUESTED** [review]
pls also add the example to `examples/readme.md`

**Comment by supernovahs** [comment]
yeah , the exex is missing . I was first trying to get an idea on how to get a custom rpc .

**Comment by gakonst** [comment]
Clarifying that we'd like the _ExEx_ to spawn the RPC in a custom port or path in the JSON-RPC, this is not a task for opening an additional JSON RPC namespace in the same port!

**Pull request closed without merging**